### PR TITLE
DEV-34 Implement JWT Authentication on the Backend

### PR DESCRIPTION
**Pull request by mpark63**
Shoutout to @QinyuanWu!!! 🔥 

## Rationale 
Currently our backend is super unsecure! Anyone can make a GET/POST/PATCH/DELETE call to it, modifying our database. We need to secure it with JWT tokens! 

[frontend PR](https://github.com/uCredit-Dev/ucredit_frontend_typescript/pull/413)

aiming to add comments explaining code changes, sometime soon.

### CODE REVIEW

File: routes/comment.js [hunk 1]
```diff
@@ -79,34 +94,49 @@ router.patch("/api/thread/resolve", (req, res) => {
...
 */
-router.delete("/api/comment", (req, res) => {
+router.delete("/api/comment", auth, (req, res) => {
   const comment_id = req.body.comment_id;
   if (!comment_id) errorHandler(res, 400, { message: "Missing comment_id." });
-  else
+  else {
+    Comments.findById(comment_id)
+      .then((comment) => {
+        // TODO: comment is null error
```
Comment (mpark63) [inline routes/comment.js:123]: todo; backend gives comment is null error

File: routes/course.js [hunk 2]
```diff
@@ -49,23 +68,29 @@ router.get("/api/courses/:course_id", (req, res) => {
     .catch((err) => errorHandler(res, 400, err));
 });
 
+// NOT_IN_USE 
 // get courses in a plan by term. provide plan id, year, and term
-router.get("/api/coursesByTerm/:plan_id", (req, res) => {
+router.get("/api/coursesByTerm/:plan_id", auth, (req, res) => {
   const plan_id = req.params.plan_id;
   const year = req.query.year;
   const term = req.query.term;
```
Comment (mpark63) [inline routes/course.js:76]: should we identify comments not in use and delete?

File: routes/notification.js [hunk 3]
```diff
@@ -19,9 +23,14 @@ router.get("/api/notifications/:user_id", (req, res) => {
...
 
+// NOT_IN_USE ?
 /* Create a notification */
-router.post("/api/notifications", (req, res) => {
+router.post("/api/notifications", auth, (req, res) => {
   const notification = req.body;
+  // verify that notification belongs to user 
+  if (!notification.user_id.includes(req.user._id)) {
+    return forbiddenHandler(res);
+  }
```
Comment (mpark63) [inline routes/notification.js:33]: .includes because notification.user_id is an array

**Review by QinyuanWu — APPROVED** [review]
approved

**Pull request merged**